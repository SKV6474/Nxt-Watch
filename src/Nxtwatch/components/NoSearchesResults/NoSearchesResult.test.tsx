import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import NoSearchResults from ".";

const handleRetrySearch = jest.fn();

describe("NoSearch Video", () => {
  beforeEach(() => {
    jest.spyOn(console, "warn").mockImplementation(() => {});
  });

  it("NoSearch Video renders correctly", () => {
    const tree = renderer
      .create(<NoSearchResults onChange={handleRetrySearch} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("No Search Retry function Search", () => {
    const { container } = render(
      <NoSearchResults onChange={handleRetrySearch} />
    );

    const NoSeachRetryElement = container.querySelector("#NoSearchRetryId");

    expect(NoSeachRetryElement).toBeInTheDocument();
  });
});
