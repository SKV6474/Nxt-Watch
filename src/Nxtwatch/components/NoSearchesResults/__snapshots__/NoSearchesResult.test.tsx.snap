// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoSearch Video NoSearch Video renders correctly 1`] = `
<div
  className="sc-fYdXmn gZsElE"
>
  <img
    alt="No Videos"
    className="sc-dxroEu kHDFbZ"
    src="https://assets.ccbp.in/frontend/react-js/nxt-watch-no-search-results-img.png "
  />
  <div
    className="sc-UxxwN ieiWmt"
  >
    No Search results found
  </div>
  <div
    className="sc-jJgynQ IRpEs"
  >
    Try different key words or remove search filter
  </div>
  <button
    className="sc-bbpGSz hdcQam"
    id="NoSearchRetryId"
    onClick={[Function]}
  >
    Retry
  </button>
</div>
`;
